src/expr/symbol_table.cpp

```diff
@@ -212,11 +213,10 @@ Expr OverloadedTypeTrie::getOverloadedFunctionForTypes(
           Trace("parser-overloading")
               << "Parametric overloaded datatype selector " << name << " "
               << tna << std::endl;
-          DatatypeType tnd = static_cast<DatatypeType>(argTypes[i]);
-          const Datatype& dt = tnd.getDatatype();
+          const DType& dt = TypeNode::fromType(argTypes[i]).getDType();
           // tng is the "generalized" version of the instantiated parametric
           // type tna
-          Type tng = dt.getDatatypeType();
+          Type tng = dt.getTypeNode().toType();
```

Comment:
I think that we will ultimately want to convert `SymbolTable` to be a term-level class (and move it to src/parser) but this is fine for now.

Reply:
Yes that's right.  This PR concerns eliminating Expr-level Datatype and not Expr-level Type, and thus is out of scope in this PR.